operator/api/redpanda/v1alpha2/redpanda_clusterspec_types.go

```diff
@@ -18,10 +18,24 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	applycorev1 "k8s.io/client-go/applyconfigurations/core/v1"
+	"k8s.io/utils/ptr"
 
 	"github.com/redpanda-data/redpanda-operator/operator/api/apiutil"
 )
 
+func (r *RedpandaClusterSpec) MarshalJSON() ([]byte, error) {
+	type RedpandaClusterSpecAlt RedpandaClusterSpec
+	alt := (*RedpandaClusterSpecAlt)(r.DeepCopy())
```

Comment:
If possible, I'd perform this casting in the `json.Marshal` call. It's a pretty funky method so limiting it's "scope" would be helpful for readers. A comment indicating what this particular [arcana](https://www.merriam-webster.com/dictionary/arcana) is would be helpful too.

Reply:
I have no objections with your suggestion. Will do exactly that

Reply:
The problem I see json.Marshal is not called inside Operator source code, but rather in gotohelm Render and LoadValues functions. That's where the casting not applicable as far as I see. I can only cast with DeepCopy before calling Render function.